src/cockpit/misc/print.py

```diff
@@ -50,6 +51,16 @@ def control(self, command: str, **kwargs: Any) -> None:
         """Send a control message, build from **kwargs"""
         self.json('', command=command, **kwargs)
 
+    def authorize(self, cookie: str, b64_content: str, auth_type: str = "Basic", **kwargs: Any) -> None:
+        """Send an authorize message with extra **kwargs"""
+        encoded_response = f"{auth_type} {b64_content}"
+        self.json('', command="authorize", cookie=cookie, response=encoded_response, **kwargs)
+
+    def basic(self, cookie: str, user: str, password: str, **kwargs: Any) -> None:
```

Comment:
I'd also take the user:pass as a single string that the user is expected to provide their own `:` for.  I feel less strongly about this one, though.